analysers/analyser_osmosis_powerline.py

```diff
@@ -24,28 +24,110 @@
 from .Analyser_Osmosis import Analyser_Osmosis
 from modules.Stablehash import stablehash64
 
-# Power lines nodes with their voltage as array padded up to 99 zeros
+# Power lines nodes with their voltage as array padded up to 99 zeros to cope with non-numerical values
 # Lines with no voltage get null voltage instead empty array
 sql01 = """
 CREATE TEMP TABLE power_lines_nodes AS
 SELECT
-    ways.id as wid,
-    unnest(ways.nodes) AS nid,
-    unnest(ways.nodes[2:array_length(ways.nodes,1)]) AS nid_next,
-    ways.tags->'cables' as cables,
-    ways.tags->'circuits' as circuits,
-    coalesce(ways.tags->'location','overhead') as location,
+    w.id as wid,
+    unnest('{NULL}' || w.nodes[1:array_length(w.nodes, 1) - 1]) AS nid_prec,
+    unnest(w.nodes) AS nid,
+    unnest(w.nodes[2:array_length(w.nodes, 1)]) AS nid_next,
+    w.tags->'cables' AS cables,
+    coalesce((w.tags->'circuits')::integer, 1) AS circuits,
+    coalesce(w.tags->'location', 'overhead') AS location,
     voltage
 FROM
-    ways
-    JOIN LATERAL (SELECT array_agg(lpad(v, 99, '0')) FROM unnest(regexp_split_to_array(ways.tags->'voltage','; *')) AS t(v)) AS t(voltage) ON TRUE
+    ways AS w
```

Comment:
Please does not add alias just to to add alias, in order to limit the code changes, and w is not so much sorted that ways.

Reply:
Again it was to ease bug investigation, it's less error prone to change one table name instead of ten other names